File: js/tests/sortByStringLength.test.js
const sortByLength = require("../sortByStringLength");

const data = [
  [
    ["a", "ccc", "dddd", "bb"],
    ["a", "bb", "ccc", "dddd"]
  ],
  [
    ["apple", "pie", "shortcake"],
    ["pie", "apple", "shortcake"]
  ],
  [
    ["may", "april", "september", "august"],
    ["may", "april", "august", "september"]
  ],
  [["maybe"], ["maybe"]],
  [[], []]
];

describe("sortByLength", () => {
  describe("Return array of strings sorted by length", () => {
    test.each(data)("sortByLength(%p) = %p", (arr, result) => {
      let actual = sortByLength(arr);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/SortByStringLengthTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class SortByStringLengthTest
    {
        [Theory]
        [InlineData(new string[] { "a", "ccc", "dddd", "bb" }, new string[] { "a", "bb", "ccc", "dddd" })]
        [InlineData(new string[] { "apple", "pie", "shortcake" }, new string[] { "pie", "apple", "shortcake" })]
        [InlineData(new string[] { "may", "april", "september", "august" }, new string[] { "may", "april", "august", "september" })]
        [InlineData(new string[] { "maybe" }, new string[] { "maybe" })]
        [InlineData(new string[] { }, new string[] { })]
        public void SortByLength_StringArrayInput_ReturnStringArraySortedByLength(string[] arr, string[] expected)
        {
            var actual = SortByStringLength.SortByLength(arr);

            Assert.Equal(expected, actual);
        }
    }
}